Common++/header/PointerVector.h

```diff
@@ -26,84 +28,130 @@ namespace pcpp
 		/**
 		 * Iterator object that is used for iterating all elements in the vector
 		 */
-		typedef typename std::vector<T*>::iterator VectorIterator;
+		using VectorIterator = typename std::vector<T*>::iterator;
 
 		/**
 		 * Const iterator object that is used for iterating all elements in a constant vector
 		 */
-		typedef typename std::vector<T*>::const_iterator ConstVectorIterator;
+		using ConstVectorIterator = typename std::vector<T*>::const_iterator;
 
 		/**
 		 * A constructor that create an empty instance of this object
 		 */
 		PointerVector()
 		{}
 
+		/**
+		 * Copies the vector along with all elements inside it.
+		 * All elements inside the copied vector are duplicates and the originals remain unchanged.
+		 * @param[in] other The vector to copy from.
+		 * @remarks As the vector is copied via deep copy, all pointers obtained from the copied vector
+		 *	reference the duplicates and not the original.
```

Comment:
is the tab at the beginning made by clang-format?

Reply:
I don't think so. I used a tab there for indentation, tho in VS it shows as a 2-space tab.
![image](https://github.com/seladb/PcapPlusPlus/assets/20269350/75a27c8b-d3a4-42bc-9071-2e2e334e16df)

Reply:
sorry, I meant between " *" and "reference".

Reply:
Yeah, I meant that one. I used a tab there, so its not from the formatter.

Reply:
I see, seems that clang-format doesn't catch it.
I think our codebase doesn't use tab in comments, could you change to spaces?